### PR TITLE
kube-proxy: fix misleading in help info

### LINKED ISSUE #105439: kube-proxy: iptables is the best-available proxy, while ipvs has advantages over iptables.
https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/README.md#ipvs-vs-iptables stipulates, that IPVS is better than IPTABLES. This is reiterated at https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies.

>$ kube-proxy --help
>…
>      --proxy-mode ProxyMode                         Which proxy mode to use: 'userspace' (older) or 'iptables' (faster) or 'ipvs' or 'kernelspace' (windows). If blank, use the best-available proxy (currently iptables). If the iptables proxy is selected, regardless of how, but the system's kernel or iptables versions are insufficient, this always falls back to the userspace proxy.

There is a contradiction: the best-available proxy currently is iptables, while IPVS has advantages over iptables.

### PR DESCRIPTION
**Pull request by jonyhy96**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #105439

### CODE REVIEW

**Comment by dilyanpalauzov** [comment]
This change is in the right direction.  But closing https://github.com/kubernetes/kubernetes/issues/105439 means to write down the advantages of iptables over ipvs.

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
...
 	fs.BoolVar(&o.config.BindAddressHardFail, "bind-address-hard-fail", o.config.BindAddressHardFail, "If true kube-proxy will treat failure to bind to a port as fatal and exit")
 	fs.Var(utilflag.PortRangeVar{Val: &o.config.PortRange}, "proxy-port-range", "Range of host ports (beginPort-endPort, single port or beginPort+offset, inclusive) that may be consumed in order to proxy service traffic. If (unspecified, 0, or 0-0) then ports will be randomly chosen.")
-	fs.Var(&o.config.Mode, "proxy-mode", "Which proxy mode to use: 'userspace' (older) or 'iptables' (faster) or 'ipvs' or 'kernelspace' (windows). If blank, use the best-available proxy (currently iptables). If the iptables proxy is selected, regardless of how, but the system's kernel or iptables versions are insufficient, this always falls back to the userspace proxy."+
+	fs.Var(&o.config.Mode, "proxy-mode", "Which proxy mode to use: 'iptables', 'ipvs', 'kernelspace' or 'userspace' (deprecated). Use iptables on linux and userspace on windows by default."+
```
Comment (danwinship) [inline cmd/kube-proxy/app/server.go:174]: How about: `"Which proxy mode to use: 'iptables' (Linux-only), 'ipvs' (Linux-only), 'kernelspace' (Windows-only), or 'userspace' (Linux/Windows, deprecated). The default value is 'iptables' on Linux and 'userspace' on Windows."`

Comment (jonyhy96) [reply cmd/kube-proxy/app/server.go:174]: This would be much more clear

**Comment by danwinship** [comment]
/approve
/lgtm
sorry such a simple fix took so long to get in...

**Pull request merged**